### PR TITLE
MM-20977 - Inviting multiple users with valid/allowed and invalid emails causes the invites for the valid users not to be sent

### PR DESCRIPTION
**Pull request by reflog**
#### Summary
fixed adding team members using new addUsersToTeamGracefully
#### Ticket Link
https://mattermost.atlassian.net/browse/MM-20977

#### Related Pull Requests
Redux: https://github.com/mattermost/mattermost-redux/pull/1025
Server: https://github.com/mattermost/mattermost-server/pull/13372 and https://github.com/mattermost/mattermost-server/pull/13630

### CODE REVIEW

File: actions/invite_actions.jsx [hunk 1]
```diff
@@ -37,9 +37,10 @@ export function sendMembersInvites(teamId, users, emails) {
             } catch (e) {
```
Comment (jespino) [inline actions/invite_actions.jsx]: In the previous line, it shouldn't be using the "graceful" version?

Comment (jespino) [reply actions/invite_actions.jsx]: nevermind, I saw the change later

File: actions/invite_actions.jsx [hunk 2]
```diff
@@ -37,9 +37,10 @@ export function sendMembersInvites(teamId, users, emails) {
...
                 response = {error: localizeMessage('invite.members.unable-to-add-the-user-to-the-team', 'Unable to add the user to the team.')};
             }
+            const errors = response.data.errors;
```
Comment (jespino) [inline actions/invite_actions.jsx]: probably something like `response.data.errors || {};` what happens if no error is returned?

File: actions/invite_actions.jsx [hunk 3]
```diff
@@ -37,9 +37,10 @@ export function sendMembersInvites(teamId, users, emails) {
             } catch (e) {
                 response = {error: localizeMessage('invite.members.unable-to-add-the-user-to-the-team', 'Unable to add the user to the team.')};
             }
+            const errors = response.data ? (response.data.errors || {}) : {};
```
Comment (sudheerDev) [inline actions/invite_actions.jsx]: Show we be handling for `response.error` object as well? What happens if all users were invalid?

Comment (reflog) [reply actions/invite_actions.jsx]: @sudheerDev the new method addTeamMembersGracefully will return all errors inside the 'response.data.errors', not in the error field. So we are ignoring it

Comment (sudheerDev) [reply actions/invite_actions.jsx]: @reflog , probably i am missing something here. From the looks of it https://github.com/mattermost/mattermost-redux/pull/1010/files#diff-69b6b6dec7aa6243100f46236dad4f7eR472 returns an error object. So i am assuming there are still error only cases without `data.errors`.I thought if all users fail we return only error. Correct me if i am wrong.  

And if all errors are handled with `response.data.errors` i think we can remove the try catch above to make it a little explicit.

Comment (reflog) [reply actions/invite_actions.jsx]: Yes, all errors are handled with `response.data.errors` (see https://github.com/mattermost/mattermost-server/pull/13372/files#diff-d6e8e8eda818a9467cff95c90993b15fR594). I'll remove the try catch to make this clearer

Comment (sudheerDev) [reply actions/invite_actions.jsx]: Thanks got it, I was curious about error handling of this API as it seemed new to me. If that is the case you probably don't even need to check for `response.data`? It is a blocking one question though.

Comment (sudheerDev) [reply actions/invite_actions.jsx]: Never mind i guess we need the check for API auth issues i guess

**Comment by lindy65** [comment]
@reflog - I'll review after the server PR is merged 👍 Please ping me once done :)

**Comment by reflog** [comment]
@lindy65 - server has been merged, please test

**Comment by lindy65** [comment]
Thanks @reflog - will do!

**Comment by lindy65** [comment]
Hi @reflog 

I am not seeing the expected when inviting a valid and invalid email. I see both email addresses in the `invitations not sent` screen as per screenshot:

image

Steps:
1. Added licence to server and enabled email invitations
2. Changed team settings to accept invites only from emails with domain `mattermost.com`
3. Invited `alice@mattermost.com` and `alice@gmail.com` and both failed with the above error

Does this PR need to be updated since the server PR was merged?

**Comment by reflog** [comment]
blocked until https://github.com/mattermost/mattermost-server/pull/13630 and https://github.com/mattermost/mattermost-redux/pull/1034 are merged

**Pull request closed without merging**